processing/src/main/java/org/apache/druid/query/rowsandcols/RearrangedRowsAndColumns.java

```diff
@@ -130,16 +128,7 @@ public boolean isNull(int rowNum)
             @Override
             public Object getObject(int rowNum)
             {
-              Object value = accessor.getObject(pointers[start + rowNum]);
-              if (ColumnType.STRING.equals(getType()) && value instanceof List) {
-                // special handling to reject MVDs
-                throw new UOE(
-                    "Encountered a multi value column [%s]. Window processing does not support MVDs. "
-                    + "Consider using UNNEST or MV_TO_ARRAY.",
-                    name
-                );
-              }
-              return value;
```

Comment:
We added that originally since the `partition by` flow was separate. Now that we have a single flow that goes through the `LazilyDecoratedRowsAndColumns` layer, we don't need this check it gets enforced by [StringFrameColumnReader](https://github.com/apache/druid/blob/88d26e4541e31dd3107f120d24f9e7c21db160a4/processing/src/main/java/org/apache/druid/frame/read/columnar/StringFrameColumnReader.java#L90) layer. This is evident by the existing tests in MSQWindowTest passing.

This was the original plan: https://github.com/apache/druid/pull/17036#discussion_r1770810503